### ISSUE TITLE
Goldencobra

### ISSUE THREAD

**Issue by 5minpause** [rails-girls-summer-of-code/projects#38]
# Name and Contact of the Project Mentor:

Holger Frohloff, @5minpause, holger@ikusei.de
# Name of the Project:

Goldencobra
## URL:

https://github.com/ikusei/Goldencobra
# About the Project:

Goldencobra is a Content Management System built as a Rails Engine. It is currently used in over 30 production applications. It has been in development for over three years now.  
As Goldencobra 1.x has grown for over three years now, we have started to build Goldencobra 2.0 since the beginning of this year. This means not a big rewrite, but taking the existing project and refactoring it into microservices - where it makes sense. As the master branch is used in many projects the challenge is to make refactorings and further development that works in the context of current projects.
# Suitable for beginners?

While code can be messy and chaotic there are many parts where beginners can take existing code and write a new service that replicates the functionality. Everything they do can be tested in the real world in seconds and their effort doesn't land in a drawer after they are finished.
# What are issues/features students may work on?
- Integrating one of the many available JS editors so users of the CMS can edit articles without entering an admin part of the site.
- Investigatin solutions for translation issues on multi-language installations of Goldencobra.
- Working on continuous integration and deployment strategies for multiple (20+) production sites running the same software (Goldencobra) - if any student is into these kind of things…
- Testing strategies for Rails Engines in use with multiple projects
- UI/Design challenges in a Content Management System context (administration of many articles, media items etc)
- Whatever might be of interest to you, we could find a problem for you. :)

Disclaimer:
Goldencobra is 100% open source. We use it in every project and earn money doing it. As it's a Rails engine we usually integrate it into the client application and every problem that is solved in this client's application scope is reintegrated back into Goldencobra - if it fits the CMS.

**Comment by alicetragedy** [comment]
hi @5minpause, unfortunately we've already closed project submissions for this year (see our readme), as the application deadline for students is in two days and we need to review every submitted project before adding it to the list of approved projects. That's too bad! But we'd love to see you apply with this project next year!

**Comment by 5minpause** [comment]
Hi. 
That's just too bad. The website didn't clearly state that. That's why I thought the deadline was the 14th?!
Any way we could still make it work?

Thank you.
Holger

**Comment by alicetragedy** [comment]
Hi @5minpause, no, unfortunately not. It's true that we did not have a set deadline in the project guides, but the application deadline is valid for student applications only. Reviewing last-minute project submissions takes a lot of time we could be spending elsewhere, eg. supporting students who need help with their application. :)
This is why we've updated the README accordingly, but maybe it wasn't clear enough. Thanks for letting us know. I'll close this issue now!
